python/lsst/daf/butler/formatters/fitsExposureFormatter.py

```diff
@@ -33,6 +33,53 @@ class FitsExposureFormatter(Formatter):
 
     parameters = frozenset(("bbox", "origin"))
 
+    def readImageComponent(self, fileDescriptor, component):
+        """Read the image, mask, or variance component of an Exposure."""
+        # TODO: could be made more efficient *if* Exposure type objects
+        # held the class objects of their components.
+        full = self.readFull(fileDescriptor, fileDescriptor.parameters)
+        return fileDescriptor.storageClass.assembler().getComponent(full, component)
+
+    def readMetadata(self, fileDescriptor):
+        """Read all header metadata directly into a PropertyList.
+        """
+        from lsst.afw.image import readMetadata
+        return readMetadata(fileDescriptor.location.path)
+
+    def stripMetadata(self, metadata):
+        """Remove metadata entries that are parsed into components.
+
+        This is only called when just the metadata is requested; stripping
+        entries there forces code that wants other components to ask for those
+        components directly rather than trying to extract them from the
+        metadata manually, which is fragile.  This behavior is an intentional
+        change from Gen2.
+        """
+        # TODO: make sure this covers everything, by delegating to something
+        # that doesn't yet exist in afw.image.ExposureInfo.
+        from lsst.afw.image import bboxFromMetadata
+        from lsst.afw.geom import makeSkyWcs
+        bboxFromMetadata(metadata)  # always strips
+        makeSkyWcs(metadata, strip=True)
+
+    def readInfoComponent(self, fileDescriptor, component):
+        """Read a component held by ExposureInfo."""
+        from lsst.afw.image import LOCAL
+        from lsst.geom import Box2I, Point2I
+        parameters = dict(bbox=Box2I(minimum=Point2I(0, 0), maximum=Point2I(0, 0)), origin=LOCAL)
+        tiny = self.readFull(fileDescriptor, parameters)
+        return fileDescriptor.storageClass.assembler().getComponent(tiny, component)
+
+    def readFull(self, fileDescriptor, parameters=None):
```

Comment:
You are defining the parameters for `get` to be any parameters understood by the Exposure constructor?

Reply:
Not at the moment - just `bbox` and `origin`.  I suppose we need a place to document this at the StorageClass level.  I'm open to ideas.